ros2pkg/ros2pkg/verb/create.py

```diff
@@ -100,13 +101,23 @@ def main(self, *, args):
             if not maintainer.email:
                 maintainer.email = maintainer.name + '@todo.todo'
 
-        cpp_node_name = None
-        if args.cpp_node_name:
-            cpp_node_name = args.cpp_node_name
-            if args.cpp_node_name == args.cpp_library_name:
-                cpp_node_name = args.cpp_node_name + '_node'
-                print('[WARNING] node name can not be equal to the library name', file=sys.stderr)
-                print('[WARNING] renaming node to %s' % cpp_node_name, file=sys.stderr)
+        node_name = None
+        cpp_library_name = None
+        if args.build_type == 'ament_cmake' or args.build_type == 'cmake':
+            if args.cpp_library_name:
```

Comment:
when generalizing `cpp_library_name` to a general `library_name` this check can be performed in any case.